tests/python/unittest/test_subgraph_op.py

```diff
@@ -18,351 +18,448 @@
-    test_network_structure_5(subgraph_backend)
-    test_network_structure_6(subgraph_backend)
-    test_network_structure_7(subgraph_backend)
-
-def test_subgraph_exe():
-    _test_subgraph_exe('default')
-
-def test_subgraph_v2_exe():
-    _test_subgraph_exe('default_v2')
+        if subgraph_backend is not None:
+            check_call(_LIB.MXRemoveSubgraphPropertyOpNames(c_str(subgraph_backend)))
+            del os.environ['MXNET_SUBGRAPH_BACKEND']
+        return exe
+
+    original_exec = get_executor(sym)
+    partitioned_exec = get_executor(sym, subgraph_backend, op_names, original_exec)
+    outputs1 = original_exec.outputs
+    outputs2 = partitioned_exec.outputs
+    assert len(outputs1) == len(outputs2)
+    for i in range(len(outputs1)):
+        assert_almost_equal((outputs1[i] - outputs2[i]).abs().sum().asnumpy(), np.zeros(shape=(1,)))
+
+def set_random_inputs(exe1, input_names):
+    """Sets random values to exe1's args and auxs"""
+    for name in input_names:
+        if name in exe1.arg_dict:
+            exe1.arg_dict[name][:] = mx.nd.random.uniform(shape=exe1.arg_dict[name].shape)
+        else:
+            assert name in exe1.aux_dict
+            exe1.aux_dict[name][:] = mx.nd.random.uniform(shape=exe1.aux_dict[name].shape)
+
+def copy_inputs_between_executors(exe1, exe2, input_names):
+    """Copies values of args and auxs from exe1 to exe2"""
+    for name in input_names:
+        if name in exe2.arg_dict:
+            exe2.arg_dict[name][:] = exe1.arg_dict[name]
+        else:
+            assert name in exe2.aux_dict
+            exe2.aux_dict[name][:] = exe1.aux_dict[name]
+
+def check_subgraph_exe5(sym, subgraph_backend, op_names):
+    """Call optimize_for to trigger graph partitioning without infer shapes/types before,
+    then simple_bind and compare results of the partitioned sym and the original sym."""
+    # simple_bind
+    exe1 = sym.simple_bind(ctx=mx.current_context(), grad_req='null')
+    input_names = sym.list_inputs()
+    set_random_inputs(exe1, input_names)
+    exe1.forward()
+
+    # partition before simple_bind
+    check_call(_LIB.MXSetSubgraphPropertyOpNamesV2(c_str(subgraph_backend), mx_uint(len(op_names)),
+                                                 c_str_array(op_names)))
+    part_sym = sym.optimize_for(subgraph_backend)
+    check_call(_LIB.MXRemoveSubgraphPropertyOpNamesV2(c_str(subgraph_backend)))
+
+    exe2 = part_sym.simple_bind(ctx=mx.current_context(), grad_req='null')
+    copy_inputs_between_executors(exe1, exe2, input_names)
+    exe2.forward()
+
+    # compare outputs
+    outputs1 = exe1.outputs
+    outputs2 = exe2.outputs
+    assert len(outputs1) == len(outputs2)
+    for i in range(len(outputs1)):
+        assert_almost_equal((outputs1[i] - outputs2[i]).abs().sum().asnumpy(), np.zeros(shape=(1,)))
+
+def check_subgraph_exe6(sym, subgraph_backend, op_names):
+    """Call optimize_for to trigger graph partitioning without infer shapes/types before,
```

Comment:
lets change this comment, in this test we are inferring/shapes types

Reply:
Updated.